.github/workflows/pr-realm-js.yml

```
@@ -0,0 +1,306 @@
+        if: steps.cache-prebuild.outputs.cache-hit != 'true'
+        run: |
+          npm ci --ignore-scripts
+          npm run build:${{ matrix.variant.os }} -- --arch=${{matrix.variant.arch}}
+
+      - name: Upload prebuild artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: prebuild-${{ matrix.variant.os }}-${{ matrix.variant.arch }}
+          path: prebuilds
+
+      - name: Install the node dependencies in the test folder
+        working-directory: ./tests
+        run: |
+          npm ci
+
+      # Run the node test scripts (without sync)
+      - name: Run tests
+        if: ${{ (matrix.variant.test-node) }}
+        working-directory: ./tests
+        run: npm test
+
+      - name: Install electron test dependencies
+        if: ${{ (matrix.variant.test-electron) }}
+        working-directory: ./tests/electron
+        run: |
+          npm ci
+          npm install -S ../js
+
+      - name: Run electron tests on non-Linux
+        if: ${{ (matrix.variant.test-electron) && (matrix.variant.os != 'linux') }}
+        working-directory: ./tests/electron
+        run: |
+          npm test -- --process=main
+          npm test -- --process=render
+
+      - name: Run electron tests on Linux
+        if: ${{ (matrix.variant.test-electron) && (matrix.variant.os == 'linux') }}
+        working-directory: ./tests/electron
+        run: |
+          xvfb-run --auto-servernum npm test -- --process=main
+          xvfb-run --auto-servernum npm test -- --process=render
+
+      - name: Run react native tests
+        if: ${{ (matrix.variant.test-react) }}
+        run: |
+          bash ./scripts/ci-scripts/react-tests.sh
+
+      - name: Run react native example
+        if: ${{ (matrix.variant.test-react) }}
+        run: |
+          bash ./scripts/ci-scripts/realm-example.sh
+
+  test-react-native-ios:
+    name: Test React Native on iOS Simulator
+    runs-on: macos-latest
+    if: ${{false}}
+    needs: [build-and-test]
+    env:
+      REALM_DISABLE_SYNC_TESTS: true
+    steps:
+      # This will retrieve the realm-js github repository
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
+
+      - name: Download prebuild artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: prebuild-darwin-x64
+          path: prebuilds
+
+      - name: Download xcframework artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: xcframework-macos
+          path: react-native/ios/realm-js-ios.xcframework
+
+      - name: Cache iOS build artifacts
+        id: cache-ios-build
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-ios-build
+        with:
+          path: react-native/ios/build
+          key: build-ios-${{ hashFiles('./src/**') }}
+          restore-keys: |
+            build-ios-
+
+      - name: Setup environment
+        run: |
+          brew install watchman
```

I'd suggest sticking with the single line variant for these one-liners.

```suggestion
        run: brew install watchman
```